src/mhd_system.hpp

```diff
@@ -26,6 +30,609 @@ template <typename problem_t> class MHDSystem
+  constexpr double theta_degrees = 0.0; // degrees
+  // k = 2 pi / wave length; box length = 1, so |k| in [1, inf)
+  constexpr double num_modes = 2;
+  constexpr double k_amplitude = 2 * M_PI * num_modes;
+  // input perturbation: choose to do this via the relative denisty field in [0, 1]. remember, the linear regime is valid when this perturbation is small
+  constexpr double delta_b = 1e-6;
+
+  const double cos_theta = std::cos(theta_degrees * M_PI / 180.0);
+  const double sin_theta = std::sin(theta_degrees * M_PI / 180.0);
+
+  const double alfven_speed = bg_mag_amplitude / std::sqrt(bg_density);
+  const double bg_mag_x1 = bg_mag_amplitude * cos_theta;
+  const double bg_mag_x2 = bg_mag_amplitude * sin_theta;
+
+  const double omega = std::sqrt(std::pow(alfven_speed,2) * std::pow(k_amplitude,2) * std::pow(cos_theta,2));
+
+  const double cos_wave = std::cos(omega * time - k_amplitude * x1);
+
+  const double x1mag = bg_mag_x1;
+  const double x2mag = bg_mag_x2;
+  const double x3mag = bg_mag_amplitude * delta_b * cos_wave;
+
+  const double x1vel = 0;
+  const double x2vel = 0;
+  const double x3vel = -omega * delta_b / (sound_speed * k_amplitude * cos_theta) * cos_wave;
+  
+  if (itype == 0) {
+    std::array<const double, 3> array = {x1vel, x2vel, x3vel};
+    return array;
+  } else {
+    std::array<const double, 3> array = {x1mag, x2mag, x3mag};
+    return array;
+  }
+}
+
+template <typename problem_t>
+void MHDSystem<problem_t>::ReconstructTo(FluxDir dir, arrayconst_t &cState, array_t &lState, array_t &rState, const amrex::Box &box_cValid,
+					 const int reconstructionOrder)
+{
+  amrex::Box const &box_r = amrex::grow(box_cValid, 1);
+	amrex::Box const &box_r_x1 = amrex::surroundingNodes(box_r, static_cast<int>(dir));
+  if (reconstructionOrder == 3) {
+    // note: only box_r is used. box_r_x1 is unused.
+		switch (dir) {
+			case FluxDir::X1:
+				MHDSystem<problem_t>::template ReconstructStatesPPM<FluxDir::X1>(cState, lState, rState, box_r, box_r_x1, 1);
+				break;
+			case FluxDir::X2:
+				MHDSystem<problem_t>::template ReconstructStatesPPM<FluxDir::X2>(cState, lState, rState, box_r, box_r_x1, 1);
+				break;
+			case FluxDir::X3:
+				MHDSystem<problem_t>::template ReconstructStatesPPM<FluxDir::X3>(cState, lState, rState, box_r, box_r_x1, 1);
+				break;
+		}
+	} else if (reconstructionOrder == 2) {
+		switch (dir) {
+			case FluxDir::X1:
+				MHDSystem<problem_t>::template ReconstructStatesPLM<FluxDir::X1, SlopeLimiter::minmod>(cState, lState, rState, box_r_x1, 1);
+				break;
+			case FluxDir::X2:
+				MHDSystem<problem_t>::template ReconstructStatesPLM<FluxDir::X2, SlopeLimiter::minmod>(cState, lState, rState, box_r_x1, 1);
+				break;
+			case FluxDir::X3:
+				MHDSystem<problem_t>::template ReconstructStatesPLM<FluxDir::X3, SlopeLimiter::minmod>(cState, lState, rState, box_r_x1, 1);
```

Comment:
minmod might actually be too diffusive.

PLUTO (and ORION) use the Van Leer limiter for velocity and magnetic field (which is less diffusive): https://github.com/JohannesBuchner/PLUTO/blob/9356273b66309fd84778aa981344a2faeae9c756/Src/States/plm_coeffs.h#L114